src/treemodel/liveposition.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+CKEDITOR.define( [
+	'treemodel/position',
+	'treemodel/range',
+	'emittermixin',
+	'utils'
+], ( Position, Range, EmitterMixin, utils ) => {
+	const STICKS_TO_NEXT = 0;
+	const STICKS_TO_PREVIOUS = 1;
+
+	/**
+	 * LivePosition is a position in the Tree Model that updates itself as the tree changes. It may be used as a bookmark.
+	 * **Note:** Be very careful when dealing with LivePosition. Each LivePosition instance bind events that might
+	 * have to be unbound. Use {@link #detach} whenever you don't need LivePosition anymore.
+	 *
+	 * @class treeModel.LivePosition
+	 */
+
+	class LivePosition extends Position {
+		/**
+		 * Creates a live position.
+		 *
+		 * @see {@link treeModel.Position}
+		 * @param root
+		 * @param path
+		 * @param {Number} [stickiness] Flag representing how live position is "sticking" with their neighbour nodes.
+		 * Defaults to {@link #STICKS_TO_NEXT}. See {@link #stickiness}.
+		 * @constructor
+		 */
+		constructor( root, path, stickiness ) {
+			super( root, path );
+
+			/**
+			 * Flag representing LivePosition stickiness. LivePosition might be sticking to previous node or next node.
+			 * Whenever some nodes are inserted at the same position as LivePosition, `stickiness` is checked to decide if
+			 * LivePosition should be moved. Similar applies when a range of nodes is moved and one of it's boundary
+			 * position is same as LivePosition. Accepted values are {@link #STICKS_TO_PREVIOUS} and {@link #STICKS_TO_NEXT}.
```

Comment:
Example would be useful. For the person who have no idea what the problem is this explanation may be not enough. And it is very simple to explain with the ASCII image.